
=== Blueprint Registration Summary ===
âœ… Successful: 29
   - Authentication
   - Admin
   - Developer
   - Inventory
   - Recipes
   - Batches
   - Organization
   - Billing
   - Settings
   - Timers
   - Expiration
   - Conversion
   - Products Main
   - Product Inventory
   - Reservations
   - SKU Management
   - Product Variants
   - Product Alerts
   - Public API
   - Main API
   - Drawer Actions
   - Density Reference
   - App Routes
   - Legal Routes
   - Bulk Stock
   - Fault Log
   - Tag Manager
   - Waitlist
   - Production Planning

ðŸŽ‰ All blueprints registered successfully!
2025-09-11 18:35:49,557 WARNING:  * Debugger is active!
2025-09-11 18:35:49,992 INFO: Expiration alerts debug: expired_fifo=11, expired_products=0, expiring_fifo=4, expiring_products=0
2025-09-11 18:35:49,992 INFO: Current time: 2025-09-11 18:35:49.981804, cutoff: 2025-09-18 18:35:49.981804
2025-09-11 18:35:50,035 [INFO] app.utils.unit_utils: Getting global unit list
2025-09-11 18:35:50,035 INFO: Getting global unit list
Error getting active roles for user 2: (psycopg2.errors.InFailedSqlTransaction) current transaction is aborted, commands ignored until end of transaction block

[SQL: SELECT user_role_assignment.id AS user_role_assignment_id, user_role_assignment.user_id AS user_role_assignment_user_id, user_role_assignment.role_id AS user_role_assignment_role_id, user_role_assignment.developer_role_id AS user_role_assignment_developer_role_id, user_role_assignment.organization_id AS user_role_assignment_organization_id, user_role_assignment.is_active AS user_role_assignment_is_active, user_role_assignment.assigned_at AS user_role_assignment_assigned_at, user_role_assignment.assigned_by AS user_role_assignment_assigned_by 
FROM user_role_assignment 
WHERE user_role_assignment.user_id = %(user_id_1)s AND user_role_assignment.is_active = true]
[parameters: {'user_id_1': 2}]
(Background on this error at: https://sqlalche.me/e/20/2j85)
Error getting active roles for user 2: (psycopg2.errors.InFailedSqlTransaction) current transaction is aborted, commands ignored until end of transaction block

[SQL: SELECT user_role_assignment.id AS user_role_assignment_id, user_role_assignment.user_id AS user_role_assignment_user_id, user_role_assignment.role_id AS user_role_assignment_role_id, user_role_assignment.developer_role_id AS user_role_assignment_developer_role_id, user_role_assignment.organization_id AS user_role_assignment_organization_id, user_role_assignment.is_active AS user_role_assignment_is_active, user_role_assignment.assigned_at AS user_role_assignment_assigned_at, user_role_assignment.assigned_by AS user_role_assignment_assigned_by 
FROM user_role_assignment 
WHERE user_role_assignment.user_id = %(user_id_1)s AND user_role_assignment.is_active = true]
[parameters: {'user_id_1': 2}]
(Background on this error at: https://sqlalche.me/e/20/2j85)
Error getting active roles for user 2: (psycopg2.errors.InFailedSqlTransaction) current transaction is aborted, commands ignored until end of transaction block

[SQL: SELECT user_role_assignment.id AS user_role_assignment_id, user_role_assignment.user_id AS user_role_assignment_user_id, user_role_assignment.role_id AS user_role_assignment_role_id, user_role_assignment.developer_role_id AS user_role_assignment_developer_role_id, user_role_assignment.organization_id AS user_role_assignment_organization_id, user_role_assignment.is_active AS user_role_assignment_is_active, user_role_assignment.assigned_at AS user_role_assignment_assigned_at, user_role_assignment.assigned_by AS user_role_assignment_assigned_by 
FROM user_role_assignment 
WHERE user_role_assignment.user_id = %(user_id_1)s AND user_role_assignment.is_active = true]
[parameters: {'user_id_1': 2}]
(Background on this error at: https://sqlalche.me/e/20/2j85)
Error getting active roles for user 2: (psycopg2.errors.InFailedSqlTransaction) current transaction is aborted, commands ignored until end of transaction block

[SQL: SELECT user_role_assignment.id AS user_role_assignment_id, user_role_assignment.user_id AS user_role_assignment_user_id, user_role_assignment.role_id AS user_role_assignment_role_id, user_role_assignment.developer_role_id AS user_role_assignment_developer_role_id, user_role_assignment.organization_id AS user_role_assignment_organization_id, user_role_assignment.is_active AS user_role_assignment_is_active, user_role_assignment.assigned_at AS user_role_assignment_assigned_at, user_role_assignment.assigned_by AS user_role_assignment_assigned_by 
FROM user_role_assignment 
WHERE user_role_assignment.user_id = %(user_id_1)s AND user_role_assignment.is_active = true]
[parameters: {'user_id_1': 2}]
(Background on this error at: https://sqlalche.me/e/20/2j85)
Error getting active roles for user 2: (psycopg2.errors.InFailedSqlTransaction) current transaction is aborted, commands ignored until end of transaction block

[SQL: SELECT user_role_assignment.id AS user_role_assignment_id, user_role_assignment.user_id AS user_role_assignment_user_id, user_role_assignment.role_id AS user_role_assignment_role_id, user_role_assignment.developer_role_id AS user_role_assignment_developer_role_id, user_role_assignment.organization_id AS user_role_assignment_organization_id, user_role_assignment.is_active AS user_role_assignment_is_active, user_role_assignment.assigned_at AS user_role_assignment_assigned_at, user_role_assignment.assigned_by AS user_role_assignment_assigned_by 
FROM user_role_assignment 
WHERE user_role_assignment.user_id = %(user_id_1)s AND user_role_assignment.is_active = true]
[parameters: {'user_id_1': 2}]
(Background on this error at: https://sqlalche.me/e/20/2j85)
Error getting active roles for user 2: (psycopg2.errors.InFailedSqlTransaction) current transaction is aborted, commands ignored until end of transaction block

[SQL: SELECT user_role_assignment.id AS user_role_assignment_id, user_role_assignment.user_id AS user_role_assignment_user_id, user_role_assignment.role_id AS user_role_assignment_role_id, user_role_assignment.developer_role_id AS user_role_assignment_developer_role_id, user_role_assignment.organization_id AS user_role_assignment_organization_id, user_role_assignment.is_active AS user_role_assignment_is_active, user_role_assignment.assigned_at AS user_role_assignment_assigned_at, user_role_assignment.assigned_by AS user_role_assignment_assigned_by 
FROM user_role_assignment 
WHERE user_role_assignment.user_id = %(user_id_1)s AND user_role_assignment.is_active = true]
[parameters: {'user_id_1': 2}]
(Background on this error at: https://sqlalche.me/e/20/2j85)
Error getting active roles for user 2: (psycopg2.errors.InFailedSqlTransaction) current transaction is aborted, commands ignored until end of transaction block

[SQL: SELECT user_role_assignment.id AS user_role_assignment_id, user_role_assignment.user_id AS user_role_assignment_user_id, user_role_assignment.role_id AS user_role_assignment_role_id, user_role_assignment.developer_role_id AS user_role_assignment_developer_role_id, user_role_assignment.organization_id AS user_role_assignment_organization_id, user_role_assignment.is_active AS user_role_assignment_is_active, user_role_assignment.assigned_at AS user_role_assignment_assigned_at, user_role_assignment.assigned_by AS user_role_assignment_assigned_by 
FROM user_role_assignment 
WHERE user_role_assignment.user_id = %(user_id_1)s AND user_role_assignment.is_active = true]
[parameters: {'user_id_1': 2}]
(Background on this error at: https://sqlalche.me/e/20/2j85)
Error getting active roles for user 2: (psycopg2.errors.InFailedSqlTransaction) current transaction is aborted, commands ignored until end of transaction block

[SQL: SELECT user_role_assignment.id AS user_role_assignment_id, user_role_assignment.user_id AS user_role_assignment_user_id, user_role_assignment.role_id AS user_role_assignment_role_id, user_role_assignment.developer_role_id AS user_role_assignment_developer_role_id, user_role_assignment.organization_id AS user_role_assignment_organization_id, user_role_assignment.is_active AS user_role_assignment_is_active, user_role_assignment.assigned_at AS user_role_assignment_assigned_at, user_role_assignment.assigned_by AS user_role_assignment_assigned_by 
FROM user_role_assignment 
WHERE user_role_assignment.user_id = %(user_id_1)s AND user_role_assignment.is_active = true]
[parameters: {'user_id_1': 2}]
(Background on this error at: https://sqlalche.me/e/20/2j85)
Error getting active roles for user 2: (psycopg2.errors.InFailedSqlTransaction) current transaction is aborted, commands ignored until end of transaction block

[SQL: SELECT user_role_assignment.id AS user_role_assignment_id, user_role_assignment.user_id AS user_role_assignment_user_id, user_role_assignment.role_id AS user_role_assignment_role_id, user_role_assignment.developer_role_id AS user_role_assignment_developer_role_id, user_role_assignment.organization_id AS user_role_assignment_organization_id, user_role_assignment.is_active AS user_role_assignment_is_active, user_role_assignment.assigned_at AS user_role_assignment_assigned_at, user_role_assignment.assigned_by AS user_role_assignment_assigned_by 
FROM user_role_assignment 
WHERE user_role_assignment.user_id = %(user_id_1)s AND user_role_assignment.is_active = true]
[parameters: {'user_id_1': 2}]
(Background on this error at: https://sqlalche.me/e/20/2j85)
Error getting active roles for user 2: (psycopg2.errors.InFailedSqlTransaction) current transaction is aborted, commands ignored until end of transaction block

[SQL: SELECT user_role_assignment.id AS user_role_assignment_id, user_role_assignment.user_id AS user_role_assignment_user_id, user_role_assignment.role_id AS user_role_assignment_role_id, user_role_assignment.developer_role_id AS user_role_assignment_developer_role_id, user_role_assignment.organization_id AS user_role_assignment_organization_id, user_role_assignment.is_active AS user_role_assignment_is_active, user_role_assignment.assigned_at AS user_role_assignment_assigned_at, user_role_assignment.assigned_by AS user_role_assignment_assigned_by 
FROM user_role_assignment 
WHERE user_role_assignment.user_id = %(user_id_1)s AND user_role_assignment.is_active = true]
[parameters: {'user_id_1': 2}]
(Background on this error at: https://sqlalche.me/e/20/2j85)
2025-09-11 18:35:50,101 [DEBUG] app.utils.permissions: Permission organization.manage not allowed by tier exempt
2025-09-11 18:35:50,101 DEBUG: Permission organization.manage not allowed by tier exempt
Error getting active roles for user 2: (psycopg2.errors.InFailedSqlTransaction) current transaction is aborted, commands ignored until end of transaction block

[SQL: SELECT user_role_assignment.id AS user_role_assignment_id, user_role_assignment.user_id AS user_role_assignment_user_id, user_role_assignment.role_id AS user_role_assignment_role_id, user_role_assignment.developer_role_id AS user_role_assignment_developer_role_id, user_role_assignment.organization_id AS user_role_assignment_organization_id, user_role_assignment.is_active AS user_role_assignment_is_active, user_role_assignment.assigned_at AS user_role_assignment_assigned_at, user_role_assignment.assigned_by AS user_role_assignment_assigned_by 
FROM user_role_assignment 
WHERE user_role_assignment.user_id = %(user_id_1)s AND user_role_assignment.is_active = true]
[parameters: {'user_id_1': 2}]
(Background on this error at: https://sqlalche.me/e/20/2j85)
Error getting active roles for user 2: (psycopg2.errors.InFailedSqlTransaction) current transaction is aborted, commands ignored until end of transaction block

[SQL: SELECT user_role_assignment.id AS user_role_assignment_id, user_role_assignment.user_id AS user_role_assignment_user_id, user_role_assignment.role_id AS user_role_assignment_role_id, user_role_assignment.developer_role_id AS user_role_assignment_developer_role_id, user_role_assignment.organization_id AS user_role_assignment_organization_id, user_role_assignment.is_active AS user_role_assignment_is_active, user_role_assignment.assigned_at AS user_role_assignment_assigned_at, user_role_assignment.assigned_by AS user_role_assignment_assigned_by 
FROM user_role_assignment 
WHERE user_role_assignment.user_id = %(user_id_1)s AND user_role_assignment.is_active = true]
[parameters: {'user_id_1': 2}]
(Background on this error at: https://sqlalche.me/e/20/2j85)
Error getting active roles for user 2: (psycopg2.errors.InFailedSqlTransaction) current transaction is aborted, commands ignored until end of transaction block

[SQL: SELECT user_role_assignment.id AS user_role_assignment_id, user_role_assignment.user_id AS user_role_assignment_user_id, user_role_assignment.role_id AS user_role_assignment_role_id, user_role_assignment.developer_role_id AS user_role_assignment_developer_role_id, user_role_assignment.organization_id AS user_role_assignment_organization_id, user_role_assignment.is_active AS user_role_assignment_is_active, user_role_assignment.assigned_at AS user_role_assignment_assigned_at, user_role_assignment.assigned_by AS user_role_assignment_assigned_by 
FROM user_role_assignment 
WHERE user_role_assignment.user_id = %(user_id_1)s AND user_role_assignment.is_active = true]
[parameters: {'user_id_1': 2}]
(Background on this error at: https://sqlalche.me/e/20/2j85)
2025-09-11 18:35:50,105 [DEBUG] app.utils.permissions: Permission tags.manage not allowed by tier exempt
2025-09-11 18:35:50,105 DEBUG: Permission tags.manage not allowed by tier exempt
2025-09-11 18:35:50,105 [DEBUG] app.utils.permissions: Permission dev.system_admin not allowed by tier exempt
2025-09-11 18:35:50,105 DEBUG: Permission dev.system_admin not allowed by tier exempt
Traceback (most recent call last):
  File "/home/runner/workspace/.pythonlibs/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 1964, in _exec_single_context
    self.dialect.do_execute(
  File "/home/runner/workspace/.pythonlibs/lib/python3.11/site-packages/sqlalchemy/engine/default.py", line 945, in do_execute
    cursor.execute(statement, parameters)
psycopg2.errors.InFailedSqlTransaction: current transaction is aborted, commands ignored until end of transaction block


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/runner/workspace/.pythonlibs/lib/python3.11/site-packages/flask/app.py", line 1536, in __call__
    return self.wsgi_app(environ, start_response)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/workspace/.pythonlibs/lib/python3.11/site-packages/flask/app.py", line 1514, in wsgi_app
    response = self.handle_exception(e)
               ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/workspace/.pythonlibs/lib/python3.11/site-packages/flask/app.py", line 1511, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/workspace/.pythonlibs/lib/python3.11/site-packages/flask/app.py", line 919, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/workspace/.pythonlibs/lib/python3.11/site-packages/flask/app.py", line 917, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/workspace/.pythonlibs/lib/python3.11/site-packages/flask/app.py", line 902, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/workspace/.pythonlibs/lib/python3.11/site-packages/flask_login/utils.py", line 290, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/workspace/app/routes/app_routes.py", line 60, in dashboard
    return render_template("dashboard.html",
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/workspace/.pythonlibs/lib/python3.11/site-packages/flask/templating.py", line 150, in render_template
    return _render(app, template, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/workspace/.pythonlibs/lib/python3.11/site-packages/flask/templating.py", line 131, in _render
    rv = template.render(context)
         ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/workspace/.pythonlibs/lib/python3.11/site-packages/jinja2/environment.py", line 1295, in render
    self.environment.handle_exception()
  File "/home/runner/workspace/.pythonlibs/lib/python3.11/site-packages/jinja2/environment.py", line 942, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "/home/runner/workspace/app/templates/dashboard.html", line 1, in top-level template code
    {% extends 'layout.html' %}
  File "/home/runner/workspace/app/templates/layout.html", line 310, in top-level template code
    {% block content %}{% endblock %}
  File "/home/runner/workspace/app/templates/dashboard.html", line 94, in block 'content'
    {% if recipe.variations %}
  File "/home/runner/workspace/.pythonlibs/lib/python3.11/site-packages/jinja2/environment.py", line 490, in getattr
    return getattr(obj, attribute)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/workspace/.pythonlibs/lib/python3.11/site-packages/sqlalchemy/orm/attributes.py", line 566, in __get__
    return self.impl.get(state, dict_)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/workspace/.pythonlibs/lib/python3.11/site-packages/sqlalchemy/orm/attributes.py", line 1086, in get
    value = self._fire_loader_callables(state, key, passive)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/workspace/.pythonlibs/lib/python3.11/site-packages/sqlalchemy/orm/attributes.py", line 1121, in _fire_loader_callables
    return self.callable_(state, passive)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/workspace/.pythonlibs/lib/python3.11/site-packages/sqlalchemy/orm/strategies.py", line 978, in _load_for_state
    return self._emit_lazyload(
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/workspace/.pythonlibs/lib/python3.11/site-packages/sqlalchemy/orm/strategies.py", line 1141, in _emit_lazyload
    result = session.execute(
             ^^^^^^^^^^^^^^^^
  File "/home/runner/workspace/.pythonlibs/lib/python3.11/site-packages/sqlalchemy/orm/session.py", line 2365, in execute
    return self._execute_internal(
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/workspace/.pythonlibs/lib/python3.11/site-packages/sqlalchemy/orm/session.py", line 2251, in _execute_internal
    result: Result[Any] = compile_state_cls.orm_execute_statement(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/workspace/.pythonlibs/lib/python3.11/site-packages/sqlalchemy/orm/context.py", line 306, in orm_execute_statement
    result = conn.execute(
             ^^^^^^^^^^^^^
  File "/home/runner/workspace/.pythonlibs/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 1416, in execute
    return meth(
           ^^^^^
  File "/home/runner/workspace/.pythonlibs/lib/python3.11/site-packages/sqlalchemy/sql/elements.py", line 523, in _execute_on_connection
    return connection._execute_clauseelement(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/workspace/.pythonlibs/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 1638, in _execute_clauseelement
    ret = self._execute_context(
          ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/workspace/.pythonlibs/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 1843, in _execute_context
    return self._exec_single_context(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/workspace/.pythonlibs/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 1983, in _exec_single_context
    self._handle_dbapi_exception(
  File "/home/runner/workspace/.pythonlibs/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 2352, in _handle_dbapi_exception
    raise sqlalchemy_exception.with_traceback(exc_info[2]) from e
  File "/home/runner/workspace/.pythonlibs/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 1964, in _exec_single_context
    self.dialect.do_execute(
  File "/home/runner/workspace/.pythonlibs/lib/python3.11/site-packages/sqlalchemy/engine/default.py", line 945, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.InternalError: (psycopg2.errors.InFailedSqlTransaction) current transaction is aborted, commands ignored until end of transaction block

[SQL: SELECT recipe.id AS recipe_id, recipe.name AS recipe_name, recipe.instructions AS recipe_instructions, recipe.label_prefix AS recipe_label_prefix, recipe.qr_image AS recipe_qr_image, recipe.parent_id AS recipe_parent_id, recipe.is_locked AS recipe_is_locked, recipe.predicted_yield AS recipe_predicted_yield, recipe.predicted_yield_unit AS recipe_predicted_yield_unit, recipe.allowed_containers AS recipe_allowed_containers, recipe.created_by AS recipe_created_by, recipe.organization_id AS recipe_organization_id 
FROM recipe 
WHERE %(param_1)s = recipe.parent_id]
[parameters: {'param_1': 7}]
(Background on this error at: https://sqlalche.me/e/20/2j85)
